domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1894,7 +1894,7 @@ public TotaledResults(RowSet<Row> set, Integer estimatedTotal) {
                             Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
     // Start a transaction that we need to close.
     // If a transaction is already running we don't need to close it.
-    final Transaction transaction = connection.tx == null ? connection.conn.begin() : null;
+    final Transaction transaction = connection.tx == null ? connection.conn.begin().result() : null;
```

Comment:
begin() is a Future. It takes some time until the database has responded to the begin request.
begin().result() will be null.
You need some async handling, for example begin().map(...) or begin().compose(...)

Reply:
done.